exporter/opentelemetry-exporter-richconsole/src/opentelemetry/exporter/richconsole/__init__.py

```diff
@@ -173,14 +173,15 @@ def spans_to_tree(spans: typing.Sequence[ReadableSpan]) -> Dict[str, Tree]:
         trees = {}
         parents = {}
         spans = list(spans)
+        span_ids = {s.context.span_id for s in spans}
         while spans:
             for span in spans:
-                if not span.parent:
+                if not span.parent or span.parent.span_id not in span_ids:
                     trace_id = opentelemetry.trace.format_trace_id(
                         span.context.trace_id
                     )
-                    trees[trace_id] = Tree(label=f"Trace {trace_id}")
-                    child = trees[trace_id].add(
+                    tree = trees.setdefault(trace_id, Tree(label=f"Trace {trace_id}"))
```

Comment:
Avoid recreating the trace tree as now we can enter this branch for multiple spans.